src/components/DatePicker/DatePicker.tsx

```diff
@@ -10,14 +10,42 @@ import { DatePickerTrigger } from './DatePickerTrigger';
 import { IconCaretLeft, IconCaretLeftDouble, IconCaretRight, IconCaretRightDouble } from '@foundation/Icon';
 import { Validation } from '@utilities/validation';
 
+type SingleDatePickerProps = {
+    variant?: 'single';
+    onChange: (date: Date | null) => void;
+    selectsRange?: false;
```

Comment:
is selectsRange property necessary? Is is not just enough to set it's value if variant is "Single" or "Range"

Reply:
If it's not specified here, we wouldn't be able to assign it by destructuring props and we would need to check if it exists before assigning. Something like this: `const selectsRange = restProps.selectRange ?? false;` Specifying it in the type definition seems better. Also helps to clarify the various combinations of props that are accepted.

Reply:
A possible improvement would be `selectsRange?: undefined;`, stating that this prop is expected to be undefined if `variant` is `'single'`, while not breaking the destructuring assignment.

Reply:
What I mean is that I don't think this property is necessary for in the DatePicker type as it's not really something you should change or control. If your variant is "Single", then selectRange must be false and vice versa. In the DatePickerComponent which the third party one, I would set the selectRange value depending on the variant. I'm not sure why we are exposing it. 
What if f I set variant to "Single" and selectsRange to true? 🤔

Reply:
You can't because of the type restrictions :) But you're right! IT can be controlled via the variant prop:
`selectsRange={variant === 'range' ? true : false}`